src/preimages.jl

```diff
@@ -39,39 +39,68 @@ function last_overlapping(y, a)
     searchsortedfirst(y, Interval(a).hi, by=x->Interval(x).lo) - 1
 end
 
-function Contractors.preimage(y, br::Branch, X; ϵ, max_iter)
-    return root( x->br.f(x)-y, br.fprime, X; ϵ, max_iter)
-end 
+#function Contractors.preimage(y, br::Branch, X; ϵ, max_iter)
+#    return root( x->br.f(x)-y, br.fprime, X; ϵ, max_iter)
+#end 
+
+range_estimate_monotone(f, X) = hull(f(Interval(X.lo)), f(Interval(X.hi)))
 
-function Contractors.preimage(y::Interval, br::Branch, X; ϵ, max_iter)
-    y = intersect(y, hull(br.Y[1], br.Y[2]))
-    x_lo = root( x->br.f(x)-y.lo, br.fprime, X; ϵ, max_iter)
-    x_hi = root( x->br.f(x)-y.hi, br.fprime, X; ϵ, max_iter)
+function preimage_monotone(y::Interval, br::Branch, X; ϵ, max_iter)
+    f = x-> br.f(x)
+    f′(x) = br.fprime(x)
     
-    # TODO (Isaia): I'm not convinced; I think that instead of 
-    # intersecting with the image of the branch, I should treat the 
-    # case when one of the preimages is empty and work consequently,
-    # as in the commented code below.
-    # The tests are passing, but I'm not sure this is the best approach.
+    Y = intersect(range_estimate_monotone(f, X), y)
+    @debug "range_estimate" range_estimate_monotone(f, X) 
+    @debug "range ∩ y" Y
+
+    if isempty(Y)
+        return Interval(∅)
+    end
+    
+    f =x -> br.f(x)-Interval(Y.lo)  
+    x_lo = Contractors.root(f, f′, X; ϵ, max_iter)
+    @debug "x_lo", x_lo
+    
+    f = x-> br.f(x)-Interval(Y.hi)
```

Comment:
Cheap optimization: if `Y` is a point interval (`Y.lo==Y.hi`), then we do not need to compute two preimages, one is enough. This should improve performance by a factor 2 on many of our examples, since our endpoints are thin. On Lorenz, this won't do as much, since we work on the 3rd iterate and call preimages several times recursively, but at least we save something on the first call.